Gd 121

Barmalej24 opened the pull request.

Barmalej24 committed:
dev

Barmalej24 committed:
Fix

Barmalej24 committed:
Change ExceptionString

Barmalej24 committed:
fix TypeRepository

Barmalej24 committed:
fixed Repository and Interface Repository

Barmalej24 committed:
refactoring

Barmalej24 committed:
fix

Barmalej24 committed:
Fix Add/Update/Delete func in repository
Refactoring

Barmalej24 committed:
remove commented code

Barmalej24 committed:
Removed unused constructors from exceptions
Changed the entity structure
Made changes to namespice

Barmalej24 committed:
Transfer interface from domain layer
Rename Interface, Namespace

Barmalej24 committed:
delete mig

Barmalej24 committed:
fix exceptions
fix validate

Barmalej24 committed:
fix

Barmalej24 committed:
Added Validation request models for controllers
Move interfaces from application layer

Barmalej24 committed:
Project reconfiguration (interface transfer)
Input refinements in base entities
Changing return results in controllers

Barmalej24 committed:
fin naming2

Barmalej24 committed:
Fixed Naming
Moved FluentValidation to Middleware

Barmalej24 committed:
Fixed Add/Update in TemplateService

Barmalej24 committed:
fixed naming
fixed exception

Barmalej24 committed:
added user entity
modify domain entities

Barmalej24 committed:
changed Add's Functions

Barmalej24 committed:
fixed EF domain entities

Barmalej24 committed:
removed unused domain exceptions

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/Base/IModifyEntity.cs:
```diff
@@ -0,0 +1,25 @@
+    /// </summary>
+    /// <typeparam name="TEntity"> Тип полей пользователей. </typeparam>
+    /// <typeparam name="TKey"> Тип первичного ключа. </typeparam>
+    public interface IModifyEntity<TEntity, out TKey> : IEntity<TKey> where TEntity : class where TKey : struct
```
Здесь несколько where поэтому их лучше каждую с новой строки написать

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/Message.cs:
```diff
@@ -0,0 +1,81 @@
+        /// <summary>
+        /// Длинна строки описывающий направление отправки
+        /// </summary>
+        public const int MAX_DIRECTION_LENG = 10;
```
Не исправлено слово "длина"

Barmalej24 replied:
возможно реализую)

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageTemplate.cs:
```diff
@@ -0,0 +1,176 @@
+        /// <summary>
+        /// Пользователь создавший шаблон сообщения
+        /// </summary>
+        public User CreatedUser { get; }
```
Сейчас получается имя "созданный пользователь".
Единственное правильное имя, которое приходит на ум это Creator (создатель).
Возможно лучше оставить текущее имя

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageTemplate.cs:
```diff
@@ -0,0 +1,176 @@
+        /// <summary>
+        /// Пользователь изменивший шаблон сообщения
+        /// </summary>
+        public User? ModifiedUser { get; private set; }
```
Сейчас получается имя "измененный пользователь".
Единственное правильное имя, которое приходит на ум это Modifier (модификатор).
Возможно лучше оставить текущее имя

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageTemplate.cs:
```diff
@@ -0,0 +1,176 @@
+        /// <summary>
+        /// Дата изменения шаблона сообщения
+        /// </summary>
+        public DateTime? ModifiedDate { get; private set; }
```
Правильнее ModificationDate

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageTemplate.cs:
```diff
@@ -0,0 +1,176 @@
+
+            if (string.IsNullOrWhiteSpace(language))
+            {
+                throw new MessageTemplateLanguageNullOrEmptyException(ExceptionMessages.ERROR_LANG_CODE);
```
Здесь тоже нужно передавать в исключение само значение language, чтобы можно было как-то понять null там или пустая строка, т.к. эти ситуации как правило вызваны разными ошибками

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageType.cs:
```diff
@@ -0,0 +1,142 @@
+        /// <summary>
+        /// Дата изменения шаблона сообщения
+        /// </summary>
+        public DateTime? ModifiedDate { get; private set; }
```
Правильнее ModificationDate

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageType.cs:
```diff
@@ -0,0 +1,142 @@
+        /// <param name="creationDate">дата и время создания типа сообщения</param>
+        /// <param name="modifiedUserName">пользователь изменивший тип сообщения</param>
+        /// <param name="modifiedDate">дата и время изменения типа сообщения</param>
+        public MessageType(string name, bool isRemoved, User createdUserName, DateTime creationDate, User? modifiedUserName, DateTime? modifiedDate)
```
createdUserName, modifiedUserName - в текущей версии имена аргументов неправильные.
Видимо забыли изменить

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageType.cs:
```diff
@@ -0,0 +1,142 @@
+        /// <param name="isRemoved">признак удаления типа сообщения</param>
+        /// <param name="modifiedUserName">пользователь изменивший тип сообщения</param>
+        /// <param name="modifiedDate">дата и время изменения типа сообщения</param>
+        public void Update(string name, bool isRemoved, User modifiedUserName, DateTime modifiedDate)
```
modifiedUserName - имя неправильное

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageType.cs:
```diff
@@ -0,0 +1,142 @@
+        {
+            ValidateUserName(modifiedUserName);
+
+            IsRemoved = true;
```
IsRemoved - лучше назвать IsDeleted, чтобы было единообразно с названием метода Delete.
Я тоже думал, как лучше назвать у себя, и пришел, что разу уж у нас нет настоящего удаления (moved - перемещения) из БД, то Delete как-то лучше смотрится

Barmalej24 replied:
спорно, нереализуемо

m-a-ilnitsky replied:
Спорно - да.
Нереализуемо - в смысле?

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/MessageType.cs:
```diff
@@ -0,0 +1,142 @@
+
+            if (name.Length > MAX_NAME_LENGTH)
+            {
+                throw new MessageTypeNameLengthException(ExceptionMessages.ERROR_TYPE_NAME_LENGTH, name.Length.ToString());
```
Лучше ещё само имя тоже отправлять в исключение, либо только имя.
Тогда по нему можно будет хотя бы вручную посчитать количество символов.
И из самого имени сразу будет видно почему строка оказалась длиннее чем нужно

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/User.cs:
```diff
@@ -0,0 +1,56 @@
+        /// <summary>
+        /// Получает имя пользователя.
+        /// </summary>
+        public string UserName { get; }
```
У Николая и Константина этот никнэйм записан как одно слово Username, поэтому лучше сделать единообразно с ними.
Если конечно это то самое имя пользователя

Barmalej24 replied:
реализуемо

m-a-ilnitsky commented on NotificationMicroservice.Domain/Entities/User.cs:
```diff
@@ -0,0 +1,56 @@
+        /// <summary>
+        /// Получает имя пользователя.
+        /// </summary>
+        public string UserName { get; }
+
+        /// <summary>
+        /// Получает ФИО пользователя.
+        /// </summary>
+        public string FullName { get; }
+
+        /// <summary>
+        /// Получает адрес электронной почты пользователя.
+        /// </summary>
+        public string Email { get; }
```
По идее UserName, FullName, Email должны быть ValueObject-ами

Barmalej24 replied:
нет, так как они уже проваледированы двумя микросервисами до меня

oleggolen replied:
Это однозначано должно быть ValueObjects, так как домен ничего абсолютно не знает про слои выше и полнотью от них независим, и в нём должно быть всё  доменных терминах и правилах, ValueObjets это не только про валидацию, это ещё и чать umbiquitous language, на чём и держится весь DDD

m-a-ilnitsky commented on NotificationMicroservice.Domain/Exception/Message/MessageDirectionNullOrEmptyException.cs:
```diff
@@ -0,0 +1,12 @@
+    /// </summary>
+    /// <param name="message">Информационное сообщение</param>
+    /// <param name="value">значение параметра вызвашвего исключение</param>
+    public class MessageDirectionNullOrEmptyException(string message, string value) : ArgumentNullException(value, message)
```
Про эти value в исключениях я уже много писал, если нет желания это исправлять, то больше писать не буду))

m-a-ilnitsky commented on NotificationMicroservice.Application/Model/Type/TypeModel.cs:
```diff
@@ -0,0 +1,15 @@
+        public UserModel CreatedUser { get; init; } = new UserModel();
+        public DateTime CreationDate { get; init; }
+        public UserModel? ModifiedUser { get; init; } = new UserModel();
+        public DateTime? ModifiedDate { get; init; }
```
Правильнее ModificationDate

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Application/Model/User/UserModel.cs:
```diff
@@ -0,0 +1,11 @@
+    public class UserModel
+    {
+        public Guid Id { get; init; }
+        public string UserName { get; init; }
```
Если это тот же самый Username, который используется у Николая и Константина, то лучше писать одним словом Username, чтобы было единообразно

Barmalej24 replied:
реализуемо

m-a-ilnitsky commented on NotificationMicroservice.DataAccess/Repository/MessageTemplateRepository.cs:
```diff
@@ -0,0 +1,85 @@
+        /// <returns>Идентификатор добавленного шаблона сообщения.</returns>
+        public async Task<Guid> AddAsync(MessageTemplate entity, CancellationToken cancellationToken)
+        {
+            context.Entry(entity.Type).State = EntityState.Modified;
```
Это приходится делать из-за того что первичный ключ задан, до добавления в базу?
Или без этого тоже будет работать?

Barmalej24 replied:
нет, это из за того, что сущность типа создана и ее создавать повторно не нужно

m-a-ilnitsky commented on NotificationMicroservice.DataAccess/Repository/MessageTemplateRepository.cs:
```diff
@@ -0,0 +1,85 @@
+        /// <param name="entity">Сущность шаблона сообщения для добавления.</param>
+        /// <param name="cancellationToken">Токен отмены операции.</param>
+        /// <returns>Идентификатор добавленного шаблона сообщения.</returns>
+        public async Task<Guid> AddAsync(MessageTemplate entity, CancellationToken cancellationToken)
```
entity - лучше назвать template, в коде этого метода это будет выглядеть информативнее

Barmalej24 replied:
нет тут именно ентити а не дто

m-a-ilnitsky replied:
Я так и понял, сущность по имени template, вроде ничто что мешает ее так и называть.
Разве что этим словом обозначается что-то другое

m-a-ilnitsky commented on NotificationMicroservice.DataAccess/Repository/MessageTemplateRepository.cs:
```diff
@@ -0,0 +1,85 @@
+        /// <returns>Возвращает <c>true</c>, если обновление прошло успешно.</returns>
+        public async Task<bool> UpdateAsync(MessageTemplate entity, CancellationToken cancellationToken)
+        {
+            context.Entry(entity).State = EntityState.Modified;
+            context.Entry(entity.Type).State = EntityState.Modified;
+            context.Types.Attach(entity.Type);
+            context.Entry(entity.ModifiedUser).State = EntityState.Modified;
+            context.Users.Attach(entity.ModifiedUser);
```
Почему нельзя использовать метод Update от DbContext?

Barmalej24 replied:
ну у меня 2 дня ушло на этот кусок кода в 5 строк чтобы не ловить эксепшен от EF по поводу вложенных сущностей при их множественных вариации использования

m-a-ilnitsky reviewed (dismissed):
Лучше уже вмёрджить, а потом допиливать в маленьких ветках

m-a-ilnitsky commented on NotificationMicroservice.Application/Services/Abstractions/Base/IModifyService.cs:
```diff
@@ -0,0 +1,31 @@
+        /// </summary>
+        /// <param name="messageTemplate">Модель для изменения</param>
+        /// <returns>Булевое значение</returns>
+        Task<bool> UpdateAsync(TEntityEdit messageTemplate);
```
UpdateAsync и TEntityEdit - лучше использовать одинаковый глагол, либо Update, либо Edit в обоих этих именах dto

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Application/Services/Abstractions/Base/IService.cs:
```diff
@@ -0,0 +1,34 @@
+        /// </summary>
+        /// <param name="entityCreate">Сущность, используемая для создания новой записи.</param>
+        /// <returns>Идентификатор новой сущности типа <typeparamref name="TKey"/>.</returns>
+        Task<TKey?> AddAsync(TEntityCreate messageTemplate);
```
AddAsync и TEntityCreate - лучше использовать одинаковый глагол Add/Create в обоих этих именах и в именах dto

Barmalej24 replied:
возможно

m-a-ilnitsky commented on NotificationMicroservice.Application/Services/Abstractions/Base/IService.cs:
```diff
@@ -0,0 +1,34 @@
+    /// <typeparam name="TEntity">Тип сущности, с которой производится работа.</typeparam>
+    /// <typeparam name="TEntityCreate">Тип сущности, используемый для создания новых записей.</typeparam>
+    /// <typeparam name="TKey">Тип ключа сущности.</typeparam>
+    public interface IService<TEntity, TEntityCreate, TKey> 
```
TEntityCreate - лучше добавить предлог for: TEntityForCreate

Barmalej24 replied:
-

m-a-ilnitsky commented on NotificationMicroservice.Application/Services/Abstractions/Base/IModifyService.cs:
```diff
@@ -0,0 +1,31 @@
+    /// <typeparam name="TEntityEdit">Модель для изменения</typeparam>
+    /// <typeparam name="TEntityDelete">Модель для удаления</typeparam>
+    /// <typeparam name="TKey"></typeparam>
+    public interface IModifyService<TEntity, TEntityCreate, TEntityEdit, TEntityDelete, TKey> : IService<TEntity, TEntityCreate, TKey> 
```
TEntityCreate, TEntityEdit, TEntityDelete - лучше добавить предлог For после Entity

Barmalej24 replied:
-

m-a-ilnitsky commented on NotificationMicroservice.Application/Mapper/ApplicationProfile.cs:
```diff
@@ -0,0 +1,21 @@
+﻿using AutoMapper;
```
В целом лучше еще применять автоформатирование солюшена:
Analyse > Code Cleanup > Run ...
В текущей версии это действие приводит к изменению 13 файлов.
Хотя в основно здесь получилось удаление лишних using (впрочем это зависит от настроек)

Barmalej24 replied:
+

NikeTorchwood reviewed (dismissed):
Ждем мердж, все нравится

oleggolen commented on NotificationMicroservice/Controllers/MessageController.cs:
```diff
@@ -0,0 +1,42 @@
+    [Route("[controller]/[action]")]
+    public class MessageController : ControllerBase
+    {
+        private readonly IMessageApplicationService _messageService;
+        private readonly IMapper _mapper;
```
Использовать primary конструктор

Barmalej24 replied:
добавлен

oleggolen commented on NotificationMicroservice/Controllers/MessageTemplateController.cs:
```diff
@@ -0,0 +1,81 @@
+    [ApiController]
+    [Route("[controller]/[action]")]
+    public class MessageTemplateController : ControllerBase
+    {
+        private readonly ITemplateApplicationService _messageTemplateService;
+        private readonly IMapper _mapper;
```
Использовать primary конструктор

Barmalej24 replied:
добавлен

oleggolen commented on NotificationMicroservice/Controllers/MessageTemplateController.cs:
```diff
@@ -0,0 +1,81 @@
+        {
+            var template = _mapper.Map<EditTemplateModel>(request);
+            template.Id = id;
+
```
Лишний пробел

Barmalej24 replied:
удален

oleggolen commented on NotificationMicroservice/Controllers/MessageTemplateController.cs:
```diff
@@ -0,0 +1,81 @@
+            {
+                return BadRequest("Template can not be created");
+            }
+
```
Лишний пробел

Barmalej24 replied:
удален

oleggolen commented on NotificationMicroservice/Controllers/MessageTemplateController.cs:
```diff
@@ -0,0 +1,81 @@
+        {
+            var template = _mapper.Map<DeleteTemplateModel>(request);
+            template.Id = id;
+
```
Лишний пробел

Barmalej24 replied:
удален

oleggolen commented on NotificationMicroservice/Controllers/MessageTypeController.cs:
```diff
@@ -0,0 +1,82 @@
+        {
+            var type = _mapper.Map<EditTypeModel>(request);
+            type.Id = id;
+
```
Лишний пробел

Barmalej24 replied:
удален

oleggolen commented on NotificationMicroservice/Controllers/MessageTypeController.cs:
```diff
@@ -0,0 +1,82 @@
+        {
+            var type = _mapper.Map<DeleteTypeModel>(request);
+            type.Id = id;
+
```
Лишний пробел

Barmalej24 replied:
удален

oleggolen commented on NotificationMicroservice/Controllers/MessageTypeController.cs:
```diff
@@ -0,0 +1,82 @@
+    [Route("[controller]/[action]")]
+    public class MessageTypeController : ControllerBase
+    {
+        private readonly ITypeApplicationService _messageTypeService;
+        private readonly IMapper _mapper;
```
Использовать primary конструктор

Barmalej24 replied:
добавлен

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageType/MessageTypeNameLengthException.cs:
```diff
@@ -0,0 +1,12 @@
+    /// </summary>
+    /// <param name="message">Информационное сообщение</param>
+    /// <param name="value">значение параметра вызвашвего исключение</param>
+    public class MessageTypeNameLengthException(string message, string value) : ArgumentOutOfRangeException(value, message)
```
Я бы здесь добавил бы информацию про минимально допустимую и максимально возможную длины сообщения

Barmalej24 replied:
добавлено

oleggolen commented on NotificationMicroservice.Domain/Exception/MessageTemplate/MessageTemplateLanguageLengthException.cs:
```diff
@@ -0,0 +1,12 @@
+    /// </summary>
+    /// <param name="message">Информационное сообщение</param>
+    /// <param name="value">значение параметра вызвашвего исключение</param>
+    public class MessageTemplateLanguageLengthException(string message, string value) : ArgumentOutOfRangeException(value, message)
```
Я бы здесь добавил более детальню информацию про минимально возможную и максимально допустимю длины

Barmalej24 replied:
добавлено

oleggolen commented on NotificationMicroservice.Domain/Exception/Message/MessageDirectionLengthException.cs:
```diff
@@ -0,0 +1,12 @@
+    /// </summary>
+    /// <param name="message">Информационное сообщение</param>
+    /// <param name="value">значение параметра вызвашвего исключение</param>
+    public class MessageDirectionLengthException(string message, string value) : ArgumentOutOfRangeException(value, message)
```
Я бы здесь добавил информацию про минимально возможную и максимально допустимые длины

Barmalej24 replied:
добавлено

oleggolen commented on NotificationMicroservice.Domain/Entities/MessageType.cs:
```diff
@@ -0,0 +1,142 @@
+        /// <summary>
+        /// Название типа сообщения
+        /// </summary>
+        public string Name { get; private set; }
```
Тоже должно быть ValueObject

Barmalej24 replied:
произволное имя, хоть цифрами

oleggolen commented on NotificationMicroservice.Application/Services/Abstractions/Base/IModifyService.cs:
```diff
@@ -0,0 +1,31 @@
+    /// <typeparam name="TEntityDelete">Модель для удаления</typeparam>
+    /// <typeparam name="TKey"></typeparam>
+    public interface IModifyService<TEntity, TEntityCreate, TEntityEdit, TEntityDelete, TKey> : IService<TEntity, TEntityCreate, TKey> 
+        where TEntity : class 
+        where TEntityCreate : class 
+        where TEntityEdit : class 
```
Лучше сделать маркерный базовый интерфейс для сервисов, чтобы не могли подкинуть любой произольный объект, даже не смотря на то, что тут принимаются записи

Barmalej24 replied:
добавлен

oleggolen commented on NotificationMicroservice.Application/Services/Abstractions/Base/IService.cs:
```diff
@@ -0,0 +1,34 @@
+    /// <typeparam name="TEntityCreate">Тип сущности, используемый для создания новых записей.</typeparam>
+    /// <typeparam name="TKey">Тип ключа сущности.</typeparam>
+    public interface IService<TEntity, TEntityCreate, TKey> 
+        where TEntity : class 
+        where TEntityCreate : class 
```
Лучше сделать маркерный базовый интерфейс для сервисов, чтобы не могли подкинуть любой произольный объект, даже не смотря на то, что тут принимаются записи

Barmalej24 replied:
добавлен

oleggolen reviewed (changes requested):
Как будто бы изменения пошли  в обратную сторону и пошёл откат  в качестве кода, что очень странно. Надо обязательно возвращаться к DDD, общему языкуу и прочем такому. Надеюь, что бсдут исправлены мои комментарии и МИхаила

Barmalej24 committed:
Added value object Email & Username
Renamed class fields (Entities, Models)
Renamed appliction Models
Changed interfaces application services
Added base interface for application models
